js/ui/grid_core/ui.grid_core.focus.js

```diff
@@ -272,16 +354,54 @@ module.exports = {
 
                     return d.promise();
                 },
-                _generateFilterPageByKey: function(key, rowData) {
-                    var dataSource = this._dataSource,
-                        filter = this._generateFilterByKey(key, "<"),
+                _concatWithCombinedFilter: function(filter) {
+                    var combinedFilter = this.getCombinedFilter();
+                    if(combinedFilter && combinedFilter.length) {
+                        filter = [[filter], "and", combinedFilter];
+                    }
+                    return filter;
+                },
+                _generateOperationFilterByKey: function(key, rowData) {
+                    var that = this,
+                        dataSource = that._dataSource,
+                        filter = that._generateFilterByKey(key, "<"),
                         sort = dataSource.sort();
+
                     if(sort) {
                         sort.slice().reverse().forEach(function(sortInfo) {
-                            filter = [[sortInfo.selector, "=", rowData[sortInfo.selector]], "and", filter];
-                            filter = [[sortInfo.selector, sortInfo.desc ? ">" : "<", rowData[sortInfo.selector]], "or", filter];
+                            var fieldName = sortInfo.selector,
+                                value = rowData[fieldName];
+
+                            if(!isDefined(value)) {
```

Comment:
This condition is wrong. For fields with null and undefined _generateFilterForUnboundColumn method is used. It is not correct. We can use the following code to get correct sort parameters columnsController.getSortDataSourceParameters(!dataSource.remoteOperations().filtering). They will be correct for unbound columns, for local and remote operations.